pinot-core/src/main/java/org/apache/pinot/core/operator/BaseOperator.java

```diff
@@ -52,4 +54,25 @@ public final T nextBlock() {
 
   // Make it protected because we should always call nextBlock()
   protected abstract T getNextBlock();
+
+  // Enforcing sub-class to implement the getOperatorName(), as they can just return a static final,
+  // as opposed to this super class calling getClass().getSimpleName().
+  public abstract String getOperatorName();
+
+  public abstract String getExplainPlanName();
+
+  @Override
+  public List<Operator> getChildOperators() {
+    return new ArrayList<>();
+  }
+
+  @Override
+  public String toExplainString() {
+    return getExplainPlanName();
+  }
+
+  @Override
```

Comment:
Why this change ?

Reply:
By default, the explain plan description for an operator is its name, unless the operator overrides `toExplainString` function. If I remember correctly this is same as what was in the original code except that I simplified things a bit.